core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp

```diff
@@ -101,8 +106,13 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::OpenMP> {
   std::enable_if_t<!std::is_same<typename Policy::schedule_type::type,
                                  Kokkos::Dynamic>::value>
   execute_parallel() const {
+#ifdef KOKKOS_COMPILER_GNU
```

Comment:
We already skip the chunk size for NVHPC (not supported) in lines 38 - 42. You could stick with `KOKKOS_OPENMP_OPTIONAL_CHUNK_SIZE` macro by checking for GNU there.

Reply:
It looks like this patch is only wanting to drop the chunk size for static scheduling.  The `KOKKOS_OPENMP_OPTIONAL_CHUNK_SIZE` is also used (12L above) with a dynamic schedule.

Reply:
True, my bad :grin:

Reply:
Yeah the dynamic schedule does not show any regression so I did not do anything for it.